homeassistant/components/netatmo/sensor.py

```diff
@@ -197,22 +197,23 @@ def _retry(_data):
 class NetatmoSensor(Entity):
     """Implementation of a Netatmo sensor."""
 
-    def __init__(self, netatmo_data, module_name, sensor_type, station):
+    def __init__(self, netatmo_data, module_info, sensor_type):
         """Initialize the sensor."""
-        self._name = "Netatmo {} {}".format(module_name, SENSOR_TYPES[sensor_type][0])
         self.netatmo_data = netatmo_data
-        self.module_name = module_name
+        module = self.netatmo_data.station_data.moduleById(mid=module_info["id"])
+        if module["type"] == "NHC":
+            self.module_name = module_info["station_name"]
+        else:
+            self.module_name = module_info["module_name"]
+        self._name = f"{DOMAIN} {self.module_name} {SENSOR_TYPES[sensor_type][0]}"
         self.type = sensor_type
-        self.station_name = station
         self._state = None
         self._device_class = SENSOR_TYPES[self.type][3]
         self._icon = SENSOR_TYPES[self.type][2]
         self._unit_of_measurement = SENSOR_TYPES[self.type][1]
-        module = self.netatmo_data.station_data.moduleByName(
-            station=self.station_name, module=module_name
-        )
         self._module_type = module["type"]
-        self._unique_id = "{}-{}".format(module["_id"], self.type)
+        self._module_id = module_info["id"]
+        self._unique_id = f"{self._module_id}-{self.type}-{self._module_type}"
```

Comment:
Does the unique_id change too? Then this is a breaking change.

Reply:
Now that you mention it, I don't think that is necessary. Thanks for your thorough reviews as always. I'll revert that and double check.